external/fv3fit/fv3fit/testing.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, Hashable, Iterable, Mapping, Optional, Union
+from ._shared import Predictor, io
+import numpy as np
+import xarray as xr
+import os
+import yaml
+
+
+@io.register("constant-output")
+class ConstantOutputPredictor(Predictor):
+    """
+    A simple predictor meant to be used for testing.
+    
+    Supports scalar and vector outputs, where the vector outputs are all
+    of the same shape and assigned a dimension name of "z".
+    """
+
+    def __init__(
+        self,
+        sample_dim_name: str,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
+    ):
+        """Initialize the predictor
+        
+        Args:
+            sample_dim_name: name of sample dimension
+            input_variables: names of input variables
+            output_variables: names of output variables
+        
+        """
+        super().__init__(
+            sample_dim_name=sample_dim_name,
+            input_variables=input_variables,
+            output_variables=output_variables,
+        )
+        self._outputs: Dict[Hashable, Union[np.ndarray, float]] = {}
+
+    def set_outputs(self, **outputs: Union[np.ndarray, float]):
+        """
+        Args:
+            outputs: column output for each name. For scalar
+                values, use float, and for column values use 1D arrays.
+        """
+        self._outputs.update(outputs)  # type: ignore
+
+    def predict(self, X: xr.Dataset) -> xr.Dataset:
+        """Predict an output xarray dataset from an input xarray dataset."""
+        n_samples = len(X[self.sample_dim_name])
+        data_vars = {}
+        for name in self.output_variables:
+            output = self._outputs[name]
+            if isinstance(output, np.ndarray):
+                array = np.repeat(output[None, :], repeats=n_samples, axis=0)
+                data_vars[name] = xr.DataArray(
+                    data=array, dims=[self.sample_dim_name, "z"]
+                )
+            else:
+                array = np.full([n_samples], float(output))
+                data_vars[name] = xr.DataArray(data=array, dims=[self.sample_dim_name])
+        if self.sample_dim_name in X:
```

Comment:
Nit:
```suggestion
        if self.sample_dim_name in X.coords:
```

Reply:
Updating alongside using `X.coords` below for consistency.